datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx

```diff
@@ -0,0 +1,366 @@
+import React, { useState } from 'react';
+import { message, Button, Modal, Select, Typography, Tag as CustomTag } from 'antd';
+import styled from 'styled-components';
+import { BookOutlined } from '@ant-design/icons';
+
+import { useGetSearchResultsLazyQuery } from '../../../graphql/search.generated';
+import { EntityType, SubResourceType, SearchResult, Tag, GlossaryTerm } from '../../../types.generated';
+import CreateTagModal from './CreateTagModal';
+import { useEntityRegistry } from '../../useEntityRegistry';
+import { IconStyleType } from '../../entity/Entity';
+import { useAddTagsMutation, useAddTermsMutation } from '../../../graphql/mutations.generated';
+import analytics, { EventType, EntityActionType } from '../../analytics';
+import { useEnterKeyListener } from '../useEnterKeyListener';
+import { StyledTag } from '../../entity/shared/components/styled/StyledTag';
+
+type AddTagsModalProps = {
+    visible: boolean;
+    onCloseModal: () => void;
+    entityUrn: string;
+    entityType: EntityType;
+    entitySubresource?: string;
+    type?: EntityType;
+};
+
+const TagSelect = styled(Select)`
+    width: 480px;
+`;
+
+const SuggestionContainer = styled.div`
+    display: 'flex',
+    flex-direction: 'row',
+    align-items: 'center',
+`;
+
+const SuggestionText = styled.span`
+    margin-left: 2px;
+    font-size: 10px;
+    line-height: 20px;
+    white-space: nowrap;
+    margin-right: 8px;
+    opacity: 1;
+    color: #434343;
+`;
+
+const CREATE_TAG_VALUE = '____reserved____.createTagValue';
+
+const NAME_TYPE_SEPARATOR = '_::_:_::_';
+
+const getSelectedValue = (rawValue: string) => {
+    const [name, type] = rawValue.split(NAME_TYPE_SEPARATOR);
+    return {
+        name,
+        type,
+    };
+};
+
+const renderTerm = (suggestion: string, icon: JSX.Element, type: string) => ({
+    value: suggestion,
+    label: (
+        <SuggestionContainer>
+            <BookOutlined style={{ marginRight: '3%' }} />
+            <SuggestionText>{suggestion}</SuggestionText>
+        </SuggestionContainer>
+    ),
+    type,
+});
+
+const renderTag = (suggestion: string, $colorHash, $color, type: string) => ({
```

Comment:
Are there any existing components we can use? If not we should create one!

Reply:
Created a new component TagPill and returned the JSX as per requirement.

Reply:
committed